gensim/corpora/dictionary.py

```diff
@@ -354,6 +356,13 @@ def load_from_text(fname):
         with utils.smart_open(fname) as f:
             for lineno, line in enumerate(f):
                 line = utils.to_unicode(line)
+                if lineno == 0:
+                    if line.strip().isdigit():
+                        # Older versions of save_as_text may not write num_docs on first line.
+                        result.num_docs = int(line.strip())
+                        continue
+                    else:
+                        logging.warning("Text does not contain num_docs on the first line.")
```

Comment:
Perhaps, but the current message doesn't help the user in any way.

What "text"? Why should it contain "num_docs"? What should the user do about it?

This should be a backward compatible change, so "errors down the pipeline" are not acceptable.